codeMetadata.go

```diff
@@ -5,6 +5,7 @@ const lengthOfCodeMetadata = 2
 const (
 	METADATA_UPGRADEABLE = 1
 	METADATA_PAYABLE     = 2
+	METADATA_READABLE    = 3
```

Comment:
Masking with 3 will also look at Upgradeable, with which it shares the byte. Should proably use 4 instead